transcom/mymove
# B-20873-INT Issue fix

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20873)
## [INT PR](https://github.com/transcom/mymove/pull/13440)

## Summary

During testing it was found that users with deleted roles were still appearing in the availableOfficeUsers array. This work updates the sql to take into account `users_roles.deleted_at`

### How to test

NOTE: I don't think you can use seeded office users for this, so login as an admin user and create some office users of SC, TOO, and TIO roles, and at least one with safety priv. Select the same transportation office for all of them (or don't, it'll just be more work for you)

run this sql to find your transportation office id (inserting your transportation office name)
`select id from transportation_offices to2 where to2."name" = 'PPPO Malmstrom AFB - USAF'`
use the ID from that in the following sql for ou.transportation_office_id
```
select ou.id, ou.first_name, ou.last_name, to2.gbloc, r.role_name, p.privilege_name from office_users ou 
join users u on u.id = ou.user_id 
join users_roles ur on ur.user_id = u.id
join roles r on r.id = ur.role_id 
join transportation_offices to2 on ou.transportation_office_id = to2.id
left join users_privileges up on u.id = up.user_id 
left join "privileges" p on p.id = up.privilege_id 
where r.role_type = 'services_counselor'
and ou.transportation_office_id = '9804dbaa-71b0-4170-8ae2-fcba1356fb7f'
and u.active = true
```

1. Login to the office app as a Services Counselor
2. Open the network tab and find the response data
3. Run the following SQL from above on your local db
4. Compare the results with the SQL results
![image](https://github.com/user-attachments/assets/0a192de1-cfce-42f3-b121-b91b30919022)
![image](https://github.com/user-attachments/assets/faaae097-cdb3-42d1-97d6-e52082b9b64b)
5. Login as an admin user and revoke the role of one of the office users listed above
6. Login refresh your page in the office app and check your response data again; make sure the office user who's role was revoked no longer appears in availableOfficeUsers
7. Do the same for TOO and TIO, replacing values in SQL query as necessary
8. Try and break
